sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/MutableDoublePointData.java

```diff
@@ -0,0 +1,127 @@
+ */
+
+package io.opentelemetry.sdk.metrics.internal.data;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.metrics.data.DoubleExemplarData;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A mutable {@link DoublePointData}
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
+ * <p>This class is not thread-safe.
+ */
+public class MutableDoublePointData implements DoublePointData {
+
+  private long startEpochNanos;
+  private long epochNanos;
+
+  private Attributes attributes = Attributes.empty();
+
+  private double value;
+  private List<DoubleExemplarData> exemplars = Collections.emptyList();
+
+  @Override
+  public double getValue() {
+    return value;
+  }
+
+  @Override
+  public long getStartEpochNanos() {
+    return startEpochNanos;
+  }
+
+  @Override
+  public long getEpochNanos() {
+    return epochNanos;
+  }
+
+  @Override
+  public Attributes getAttributes() {
+    return attributes;
+  }
+
+  @Override
+  public List<DoubleExemplarData> getExemplars() {
+    return exemplars;
+  }
+
+  /**
+   * Sets all {@link MutableDoublePointData} values based on {@code point}.
+   *
+   * @param point The point to take the values from
+   */
+  public void set(DoublePointData point) {
+    set(
+        point.getStartEpochNanos(),
+        point.getEpochNanos(),
+        point.getAttributes(),
+        point.getValue(),
+        point.getExemplars());
+  }
+
+  /** Sets all {@link MutableDoublePointData} values , besides exemplars which are set to empty. */
+  public void set(long startEpochNanos, long epochNanos, Attributes attributes, double value) {
+    set(startEpochNanos, epochNanos, attributes, value, Collections.emptyList());
+  }
+
+  /** Sets all {@link MutableDoublePointData} values. */
+  public void set(
+      long startEpochNanos,
+      long epochNanos,
+      Attributes attributes,
+      double value,
+      List<DoubleExemplarData> exemplars) {
+    this.startEpochNanos = startEpochNanos;
+    this.epochNanos = epochNanos;
+    this.attributes = attributes;
+    this.value = value;
+    this.exemplars = exemplars;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || !(o instanceof MutableDoublePointData)) {
+      return false;
+    }
+    MutableDoublePointData pointData = (MutableDoublePointData) o;
+    return startEpochNanos == pointData.startEpochNanos
+        && epochNanos == pointData.epochNanos
+        && Double.compare(value, pointData.value) == 0
+        && Objects.equals(attributes, pointData.attributes)
+        && Objects.equals(exemplars, pointData.exemplars);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(startEpochNanos, epochNanos, attributes, value, exemplars);
```

Comment:
Not sure how often hashcode is called, but the implementation of `Objects.hash` creates an array from the input. A manual version like the one generated in `AutoValue_ImmutableDoubleDataPoint` is more optimal:
```
  @Override
  public int hashCode() {
    int h$ = 1;
    h$ *= 1000003;
    h$ ^= (int) ((startEpochNanos >>> 32) ^ startEpochNanos);
    h$ *= 1000003;
    h$ ^= (int) ((epochNanos >>> 32) ^ epochNanos);
    h$ *= 1000003;
    h$ ^= attributes.hashCode();
    h$ *= 1000003;
    h$ ^= (int) ((Double.doubleToLongBits(value) >>> 32) ^ Double.doubleToLongBits(value));
    h$ *= 1000003;
    h$ ^= exemplars.hashCode();
    return h$;
  }
```

Reply:
You are correct. To fix, I've copied implementation from the AutoGenerated one.